src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/FailoverConnectionPlugin.java

```diff
@@ -180,16 +181,15 @@ public FailoverConnectionPlugin(
 
     initSettings();
 
-    this.connectionProvider = connectionProvider;
-
     if (!this.enableFailoverSetting) {
       return;
     }
 
-    AuroraTopologyService topologyService = topologyServiceSupplier.get();
-    topologyService.setPerformanceMetricsEnabled(this.gatherPerfMetricsSetting);
+    this.topologyService = topologyServiceSupplier.get();
+    if(this.topologyService instanceof  ICanCollectPerformanceMetrics) {
+      ((ICanCollectPerformanceMetrics)topologyService).setPerformanceMetricsEnabled(this.gatherPerfMetricsSetting);
```

Comment:
```suggestion
    if (this.topologyService instanceof ICanCollectPerformanceMetrics) {
      ((ICanCollectPerformanceMetrics) topologyService).setPerformanceMetricsEnabled(this.gatherPerfMetricsSetting);
```
Are we checking if topology is instance of `ICanCollectPerformanceMetrics` for future proofing if someone wants to create their own topology class?

Reply:
Implementing ICanCollectPerformanceMetrics is optional. If topology service implements it then the driver can engage it.